### DISCUSSION TITLE
How to pass dynamic columns def via props

### DISCUSSION THREAD

**Discussion by dilliDN** [KevinVandy/material-react-table#373] [Q&A]
using single MRT pass the dynamic columns and data via props.. if am passing via props am getting below error
![image](https://user-images.githubusercontent.com/43196462/218965330-4527c531-7df3-44c6-b02c-782ea2d8e04a.png)

**Comment by dilliDN** [comment]
![image](https://user-images.githubusercontent.com/43196462/218965532-ed8101af-fad6-43b0-8c0f-39c21b3137e3.png)
![image](https://user-images.githubusercontent.com/43196462/218965623-0270f3b6-3e32-46be-8267-7bd978780d1a.png)

**Reply by KevinVandy** [reply]
Are your columns ever `null` or `undefined`. That is not allowed. At least have a fallback array with something like `?? []`

**Comment by dilliDN** [comment]
Hi @KevinVandy, there is no null or undefined columns..even tried with single  column.. still getting same 
I observed one thing.. on initial render only it is showing error whenever if dummy save happen in columns then it is showing.. attached sample

**Reply by dilliDN** [reply]
![ezgif com-crop](https://user-images.githubusercontent.com/43196462/223178727-942fffd9-5276-461e-a78b-34811f5ccea4.gif)

**Reply by dilliDN** [reply]
@KevinVandy can you please look into it
